server/server.go

```diff
@@ -32,6 +32,10 @@ type Opts struct {
 	IdleTimeout time.Duration
 	Filter      filters.Filter
 	Dial        proxy.DialFunc
+
+	// OKDoesNotWaitForUpstream can be set to true in order to immediately return
+	// OK to CONNECT requests.
+	OKDoesNotWaitForUpstream bool
```

Comment:
This is an awkward name, but it allows us to preserve the default behavior when this setting is unspecified.

Reply:
I guess we don't need to keep the name convention of `proxy` package here. Maybe simply `OKImmediately` as the comment suggested?

Reply:
Yeah, that might sound a little better. @myleshorton already merged this. After we're done with all the PRs, I can just come in and change this one piece.

Reply:
I didn't notice it's already merged. No big deal anyway.